bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/codemining/LineHeaderCodeMining.java

```diff
@@ -73,4 +77,27 @@ public LineHeaderCodeMining(Position position, ICodeMiningProvider provider, Con
 		super(position, provider, action);
 	}
 
+	@Override
+	public Point draw(GC gc, StyledText textWidget, Color color, int x, int y) {
+		String title= getLabel() != null ? getLabel() : "no command"; //$NON-NLS-1$
+		String[] lines= title.split("\n"); //$NON-NLS-1$
```

Comment:
Is this safe to do? Depending on formatting, you might have stray "\r" symbols in the lines you get

Reply:
good point, I didn't expect that StyledText control support \r character as line break. Is the regex ok?